### PR TITLE
Refactor: Ceanup `Deploy()` 

### PR DESCRIPTION
**Pull request by warber**
This PR basically just takes away the burden of doing everything in one function (`Deploy()`) but split it up into multiple smaller parts

### CODE REVIEW

File: cmd/monaco/deploy/deploy.go [hunk 1]
```diff
@@ -35,98 +34,99 @@ import (
...
-		return fmt.Errorf("error while finding absolute path for `%s`: %w", deploymentManifestPath, err)
+	if deployErrs != nil {
+		printErrorReport(deployErrs)
+		return fmt.Errorf("errors during %s", getOperationNounForLogging(opts.DryRun))
 	}
+	log.Info("%s finished without errors", getOperationNounForLogging(opts.DryRun))
+	return nil
+}
+
+func absPath(manifestPath string) (string, error) {
```
Comment (jskelin) [inline cmd/monaco/deploy/deploy.go:75]: wwhy not move this function to `command.go` where is used (or create `util.go` file if you prefer to keep `command.go` clear)

File: cmd/monaco/deploy/deploy.go [hunk 2]
```diff
@@ -35,98 +34,99 @@ import (
...
+	manifestPath = filepath.Clean(manifestPath)
+	return filepath.Abs(manifestPath)
+}
 
-	manifest, errs := manifest.LoadManifest(&manifest.ManifestLoaderContext{
+func loadManifest(fs afero.Fs, manifestPath string) (*manifest.Manifest, error) {
```
Comment (jskelin) [inline cmd/monaco/deploy/deploy.go:80]: same as for `absPath`

File: cmd/monaco/deploy/deploy.go [hunk 3]
```diff
@@ -35,98 +34,99 @@ import (
...
 		errutils.PrintErrors(errs)
-		return errors.New("error while loading manifest")
+		return nil, errors.New("error while loading manifest")
 	}
 
-	environments := manifest.Environments
+	return &m, nil
+}
+
+func loadRelevantEnvironments(environments manifest.Environments, environmentGroup string, specificEnvironments []string) (manifest.Environments, error) {
```
Comment (jskelin) [inline cmd/monaco/deploy/deploy.go:94]: same as for `absPath`

File: cmd/monaco/deploy/deploy.go [hunk 4]
```diff
@@ -35,98 +34,99 @@ import (
...
 
-	environmentNames := maps.Keys(environments)
-	workingDir := filepath.Dir(absManifestPath)
-
-	apis := api.NewApis()
+	return environments, nil
+}
 
-	log.Debug("Loading configuration projects ...")
+func loadRelevantProjects(fs afero.Fs, man *manifest.Manifest, manifestPath string, specificProject []string, environmentNames []string) ([]project.Project, error) {
```
Comment (jskelin) [inline cmd/monaco/deploy/deploy.go:117]: same as for `absPath`

File: cmd/monaco/deploy/deploy.go [hunk 5]
```diff
@@ -146,47 +146,28 @@ func loadProjectsToDeploy(specificProject []string, projects []project.Project,
...
+	return sortedConfigs, nil
+}
 
-	if deploymentErrors != nil {
-		printErrorReport(deploymentErrors)
-
-		return fmt.Errorf("errors during %s", getOperationNounForLogging(opts.DryRun))
-	} else {
-		log.Info("%s finished without errors", getOperationNounForLogging(opts.DryRun))
+func logProjectsInfo(projects []project.Project) {
```
Comment (jskelin) [inline cmd/monaco/deploy/deploy.go:158]: same as for `absPath`

File: cmd/monaco/deploy/deploy.go [hunk 6]
```diff
@@ -146,47 +146,28 @@ func loadProjectsToDeploy(specificProject []string, projects []project.Project,
...
+	log.Info("Projects to be deployed:")
+	for _, p := range projects {
+		log.Info("  - %s", p)
 	}
-
-	return nil
 }
 
+func logEnvironmentsInfo(environments manifest.Environments) {
```
Comment (jskelin) [inline cmd/monaco/deploy/deploy.go:165]: you know the drill...
same as for `absPath`

**Review by jskelin — COMMENTED** [review]
i would move functions closer to caller. Another is OK

File: cmd/monaco/deploy/deploy.go (hunk 1 above)
Comment (warber) [reply cmd/monaco/deploy/deploy.go:102]: i've moved everything into deploy.go

**Pull request closed without merging**